### ISSUE TITLE
NPE on  map

### ISSUE THREAD

**Issue by Lineflyer** [cgeo/cgeo#7303]
New crash report with 2019.02.17-RC. 
Not sure if this can be related to sdk-raise or not.

Device: Android 9.0

Crash report:
```
java.lang.NullPointerException: 
 
  at io.reactivex.internal.functions.ObjectHelper.requireNonNull (ObjectHelper.java:39)
 
  at io.reactivex.Observable.combineLatest (Observable.java:433)
 
  at cgeo.geocaching.sensors.GeoDirHandler.start (GeoDirHandler.java:111)
 
  at cgeo.geocaching.sensors.GeoDirHandler.start (GeoDirHandler.java:77)
 
  at cgeo.geocaching.maps.mapsforge.v6.NewMap.initializeLayers (NewMap.java:784)
 
  at cgeo.geocaching.maps.mapsforge.v6.NewMap.onStart (NewMap.java:725)
 
  at android.app.Instrumentation.callActivityOnStart (Instrumentation.java:1392)
 
  at android.app.Activity.performStart (Activity.java:7157)
 
  at android.app.ActivityThread.handleStartActivity (ActivityThread.java:2952)
 
  at android.app.servertransaction.TransactionExecutor.performLifecycleSequence (TransactionExecutor.java:180)
 
  at android.app.servertransaction.TransactionExecutor.cycleToPath (TransactionExecutor.java:165)
 
  at android.app.servertransaction.TransactionExecutor.executeLifecycleState (TransactionExecutor.java:142)
 
  at android.app.servertransaction.TransactionExecutor.execute (TransactionExecutor.java:70)
 
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:1823)
 
  at android.os.Handler.dispatchMessage (Handler.java:107)
 
  at android.os.Looper.loop (Looper.java:198)
 
  at android.app.ActivityThread.main (ActivityThread.java:6729)
 
  at java.lang.reflect.Method.invoke (Native Method)
 
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:493)
 
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:858)
```

**Comment by Lineflyer** [comment]
There is a similar NPE reported also for 2019.02.17-RC but seems related to Google Maps rather then new map:

```
java.lang.RuntimeException: 
 
  at android.app.ActivityThread.performResumeActivity (ActivityThread.java:3844)
 
  at android.app.ActivityThread.handleResumeActivity (ActivityThread.java:3885)
 
  at android.app.ActivityThread.handleLaunchActivity (ActivityThread.java:3051)
 
  at android.app.ActivityThread.-wrap14 (ActivityThread.java)
 
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:1642)
 
  at android.os.Handler.dispatchMessage (Handler.java:102)
 
  at android.os.Looper.loop (Looper.java:154)
 
  at android.app.ActivityThread.main (ActivityThread.java:6776)
 
  at java.lang.reflect.Method.invoke (Native Method)
 
  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:1518)
 
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1408)
Caused by: java.lang.NullPointerException: 
 
  at io.reactivex.internal.functions.ObjectHelper.requireNonNull (ObjectHelper.java:39)
 
  at io.reactivex.Observable.combineLatest (Observable.java:433)
 
  at cgeo.geocaching.sensors.GeoDirHandler.start (GeoDirHandler.java:111)
 
  at cgeo.geocaching.sensors.GeoDirHandler.start (GeoDirHandler.java:77)
 
  at cgeo.geocaching.maps.CGeoMap.onResume (CGeoMap.java:528)
 
  at cgeo.geocaching.maps.google.v1.GoogleMapActivity.onResume (GoogleMapActivity.java:68)
 
  at android.app.Instrumentation.callActivityOnResume (Instrumentation.java:1277)
 
  at android.app.Activity.performResume (Activity.java:7115)
 
  at android.app.ActivityThread.performResumeActivity (ActivityThread.java:3821)
```

**Comment by Lineflyer** [comment]
This is still the major cause for crashes in production (2019.02.23) as well as beta (2019.03.15-RC, 2019.03.17-RC).

**Comment by Lineflyer** [comment]
There are several different stacktraces for this one (similar).
They all have in common either `at cgeo.geocaching.sensors.GeoDirHandler.start`.

**Comment by Lineflyer** [comment]
Reopening to wait for beta channel feedback

**Comment by okainov** [comment]
Btw today I got several crashed when resuming app on Map activity. However it seems better than before (crashed only once) #7391

**Comment by rsudev** [comment]
@okainov do you have a bit more detail on this (exact version, map in use, possibly a log snippet or crash info sent to  play)?

**Comment by okainov** [comment]
@rsudev I don't know how to get "log snippet\crash info" when I use c:geo normally in the field (not at home connected with USB debugging to my PC). If you could provide me info how can I get the logs from normal running app, next time I can provide it, no problem.

2019.04.05-NB1-1b5b96a

New mapsforge with offline maps, map activity active, having app in background and clicking c:geo icon from "Apps" sometimes gets me to the crash.

**Comment by Lineflyer** [comment]
As I can't see the original issue (related to this stracktrace) anymore on Google Play the problem described by @okainov should be something different.

@okainov 
To get logs while in the field is difficult I have to admit.
I managed to have a logcat app on my phone being able to record the ADB log but even this would require you explicitly start it before the related issue happens :(

Anyway I suggest you open a dedicated new issue, once you have any new clue on it or a reproducible pattern.
